waffle-hardhat/test/reverted.test.ts

```diff
@@ -24,12 +24,111 @@ describe('INTEGRATION: Matchers: revertedWith', () => {
 
   revertedWithTest(provider);
 
-  it('Handle custom error', async () => {
-    await waffle.provider.send('hardhat_reset', []);
-    const wallets = waffle.provider.getWallets();
+  const deploy = async () => {
+    const wallets = provider.getWallets();
     const wallet = wallets[0];
     const factory = new ContractFactory(CustomError.abi, CustomError.bytecode, wallet);
     const matchers = await factory.deploy();
-    await expect(matchers.doRevertWithCustomError()).to.be.revertedWith('CustomError(0)');
+    return matchers;
+  };
+
+  describe('Custom errors', () => {
```

Comment:
Add a comment that these test cases are not in common `revertedWithTest` because they are exclusive to hardhat. Or are they not? maybe we will make them work for L2 as well? maybe we would like something like this:

```ts
revertedWithTest(provider, {includeCustomErrors: true});
```